### PR TITLE
Sort output of SHOW <INFO> statements

### PR DESCRIPTION
**Pull request by rschu1ze**
Sorting of results of `SHOW <INFO>` (e.g. `SHOW TABLES`, `SHOW ENGINES`, ...) is strictly speaking not necessary, but
1. sorted output is more user-friendly,
2. SQL currently does not allow to sort the output of `SHOW <INFO>` otherwise, e.g. `SELECT * FROM (SHOW <INFO> ...) ORDER BY` is rejected.
3. SQL tests can take advantage of that.

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
The output of some SHOW ... statements is now sorted

### CODE REVIEW

**Review by qoega — APPROVED** [review]

**Comment by rschu1ze** [comment]
Some tests get probably destroyed by this PR, but let's see...

**Comment by qoega** [comment]
Yes. And it is a bug.

File: src/Interpreters/InterpreterShowTablesQuery.cpp [hunk 1]
```diff
@@ -51,6 +50,9 @@ String InterpreterShowTablesQuery::getRewrittenQuery()
         if (query.limit_length)
             rewritten_query << " LIMIT " << query.limit_length;
 
+        /// (*)
+        rewritten_query << " ORDER BY name";
```
Comment (qoega) [inline src/Interpreters/InterpreterShowTablesQuery.cpp:54]: Order by should be before limit by the way. 
When do we use limit?

**Comment by rschu1ze** [comment]
ClickHouse Integration Tests (asan) [1/6]
- test_keeper_map/test.py::test_create_drop_keeper_map_concurrent: #48180

ClickHouse Integration Tests (tsan) [1/6]
-test_hedged_requests/test.py::test_combination4: #48181

AST Fuzzer: #48053

**Pull request merged**